### PR TITLE
Decompiled 0x006A8388

### PR DESCRIPTION
**Pull request by Garogolun**

### CODE REVIEW

File: src/windows/footpath.c [hunk 1]
```diff
@@ -704,6 +705,48 @@ static void window_footpath_set_provisional_path_at_point(int x, int y)
...
+	RCT2_GLOBAL(RCT2_ADDRESS_MAP_SELECTION_A_Y, uint16) = y;
+	RCT2_GLOBAL(RCT2_ADDRESS_MAP_SELECTION_B_Y, uint16) = y;
+
+	RCT2_GLOBAL(RCT2_ADDRESS_MAP_ARROW_DIRECTION, uint8) = direction;
+	RCT2_GLOBAL(RCT2_ADDRESS_MAP_ARROW_X, uint16) = x;
+	RCT2_GLOBAL(RCT2_ADDRESS_MAP_ARROW_Y, uint16) = y;
+
+	int z = mapElement->base_height;
+
+	if ((mapElement->type & 0x3C) == 0) {
```
Comment (duncanspumpkin) [inline src/windows/footpath.c]: I think for most of the code we now use `map_element_get_type(type) == MAP_ELEMENT_TYPE_SURFACE` for that sort of comparison.

File: src/windows/footpath.c [hunk 2]
```diff
@@ -704,6 +705,48 @@ static void window_footpath_set_provisional_path_at_point(int x, int y)
...
+*
+*  rct2: 0x006A8388
+*/
+static void window_footpath_set_selection_start_bridge_at_point(int screenX, int screenY)
+{
+	int x, y, direction;
+	rct_map_element *mapElement;
+
+	map_invalidate_selection_rect();
+	RCT2_GLOBAL(RCT2_ADDRESS_MAP_SELECTION_FLAGS, uint16) &= ~(1 << 0) & (1 << 2);
```
Comment (duncanspumpkin) [inline src/windows/footpath.c]: This does not look correct. I think you meant  `~((1 << 0) | (1 << 2));`

File: src/windows/footpath.c (hunk 1 above)
Comment (Garogolun) [inline src/windows/footpath.c]: I did not know that function, I'll change it.

File: src/windows/footpath.c (hunk 2 above)
Comment (Garogolun) [inline src/windows/footpath.c]: Well, actually I meant `~(1 << 0) & ~(1 << 2)`. Anyhow, thank you for noticing!

**Pull request merged**